### DISCUSSION TITLE
Prometheus collector process

### DISCUSSION THREAD

**Discussion by bgmat** [dokku/dokku#4231] [Q&A]
Hi,
I want to use Prometheus to collect metrics from a Rails app.
I'm using the prometheus_exporter gem, which needs to run a collector process to aggregate metrics from all other processes.
The collector process needs to expose a port different from all the usual ports expose by the web process.
I couldn't find a way to do this running the process as another container in the app and my current assumption is that it should be implemented as a dokku plugin in order to be able to expose its own port.

I would appreciate if someone if someone could tell me if my assumption is correct and it can be done only with a plugin.

Thanks

**Comment by josegonzalez** [comment]
Could you use something like this to collect metrics on containers by label? https://github.com/stuckyhm/prometheus-docker-sd

You can use the `docker-options` plugin to add the required labels to your containers.

**Reply by bgmat** [reply]
I looked into the solution you suggested, but unfortunately it does not work for me.
I'm looking for application specific metrics, not container metrics. 
This has to work in a setup with multiple web and worker processes, where the collector process aggregates metrics from all local web & worker processes and exposes a single scrape endpoint to a remote prometheus server. 
Basically, I just need to execute that process (the collector) in a way that it is visible to the other containers (e.g. like a database) and it also exposes a webpage (like a web process)
